test/test_lib/utils.js

```diff
@@ -224,11 +224,17 @@ Utils.prototype = {
 
         let key = eventData[index];
         if (eventExpectedData[key]) {
-          if (eventExpectedData[key] instanceof BN) {
-            assert(event.args[key].eq(eventExpectedData[key]), `Event data ${key} must match the expectedData`)
-
+          if (web3.utils.isBN(eventExpectedData[key])) {
+            assert(
+              event.args[key].eq(eventExpectedData[key]),
+              `Event data ${key} must match the expectedData`
+            );
           } else {
-            assert.equal(event.args[key], (eventExpectedData[key]), `Event data ${key} must match the expectedData`);
+            assert.equal(
```

Comment:
```suggestion
            assert.strictEqual(
```
if it works with existing test cases. If not, that may also be a code smell.